src/lib/core/bidi/directionality.spec.ts

```diff
@@ -0,0 +1,120 @@
+import {async, fakeAsync, TestBed, tick} from '@angular/core/testing';
+import {Component, getDebugNode} from '@angular/core';
+import {Dir, Directionality} from './index';
+
+function initDir () {
+  document.documentElement.dir = '';
+  document.body.dir = '';
+}
+
+describe('Directionality', () => {
+  beforeEach(async(() => {
+    TestBed.configureTestingModule({
+      providers: [Directionality],
+      declarations: [TestAppWithoutDir]
+    });
+
+    TestBed.compileComponents();
+
+    initDir();
+  }));
+
+  it('should read dir from the html element if not specified on the body', () => {
+    document.documentElement.dir = 'rtl';
+
+    let fixture = TestBed.createComponent(TestAppWithoutDir);
+    let testComponent = fixture.debugElement.componentInstance;
+
+    expect(testComponent.dir.value).toBe('rtl');
+  });
+
+  it('should read dir from the body even it is also specified on the html element', () => {
+    document.documentElement.dir = 'ltr';
+    document.body.dir = 'rtl';
+
+    let fixture = TestBed.createComponent(TestAppWithoutDir);
+    let testComponent = fixture.debugElement.componentInstance;
+
+    expect(testComponent.dir.value).toBe('rtl');
+  });
+
+  it('should default to ltr if nothing is specified on either body or the html element', () => {
+    let fixture = TestBed.createComponent(TestAppWithoutDir);
+    let testComponent = fixture.debugElement.componentInstance;
+
+    expect(testComponent.dir.value).toBe('ltr');
+  });
+});
+
+describe('Dir directive', () => {
+  beforeEach(async(() => {
+    TestBed.configureTestingModule({
+      declarations: [Dir, TestAppWithDir, DirTest]
```

Comment:
FYI you shouldn't put directives you're testing directly into the `declarations` of the testing module; better to import the `NgModule` they live in for real (so `BidiModule` here)